tests/transport/best_effort_reader/publisher.cpp

```diff
@@ -0,0 +1,45 @@
+#include "AppConfig.h"
+#include "SocketWriter.h"
+#include "TestMsg.h"
+
+class Publisher {
+public:
+  Publisher(int argc, ACE_TCHAR* argv[]) : config(argc, argv) {}
+
+  int run()
+  {
+    SocketWriter sw1(AppConfig::writerId[0], config.getHostAddress());
+    SocketWriter sw2(AppConfig::writerId[1], config.getHostAddress());
+    SocketWriter sw3(AppConfig::writerId[2], config.getHostAddress());
+
+    sw1.write(TestMsg(10, "1-1"), 1);
+    sw2.write(TestMsg(10, "2-1"), 1);
+    sw3.write(TestMsg(10, "3-1"), 1);
+
+    sw1.write(TestMsg(10, "1-1 Duplicate"), 1);
+    sw2.write(TestMsg(10, "2-1 Duplicate"), 1);
+    sw3.write(TestMsg(10, "3-1 Duplicate"), 1);
+
+    sw3.write(TestMsg(10, "3-2"), 2);
+
+    sw1.write(TestMsg(99, "1-3 end"), 3);
+    sw2.write(TestMsg(99, "2-3 end"), 3);
+    sw3.write(TestMsg(99, "3-3 end"), 3);
+
+    ACE_OS::sleep(2);
```

Comment:
What is this sleep meant to do? If we need to coordinate timing between processes, we should either use an appropriate DDS API (wait_for_ack / publication_match_status == 0) or the file system to coordinate.